qiskit_experiments/library/tomography/fitters/cvxpy_utils.py

```diff
@@ -51,54 +64,168 @@ def decorated_func(*args, **kwargs):
-                        var = cvxpy.Variable((5, 5), PSD=True)
-                        obj = cvxpy.Minimize(cvxpy.norm(var))
-                        cvxpy.Problem(obj).solve(solver="SCS")
-                        cls._SDP_SOLVERS.add("SCS")
-                        cls._HAS_SDP_SOLVER = True
-                    except cvxpy.error.SolverError:
-                        pass
+def solve_iteratively(problem: Problem, iters: int, **solve_kwargs) -> None:
+    """Solve a CVXPY problem increasing iterations if solution is inaccurate.
+
+    If the problem is not solved with the initial ``iters`` value of
+    iterations the number of iterations will be doubled up to the
+    specified ``"max_iters"`` value in the solve_kwargs. If no max
+    iters is specified this will be set to 4 times the initial iters
+    values.
+
+    Args:
+        problem: A CVXPY Problem to solve
+        iters: The initial number of max iterations to use when solving
+               the problem
+        solve_kwargs: kwargs for problem.solve method.
+
+    Raises:
+        AnalysisError: If the CVXPY solve fails to return an optimal or
+                       optimal_inaccurate solution.
+    """
+    max_iters = solve_kwargs.get("max_iters", 4 * iters)
+    problem_solved = False
+    while not problem_solved:
+        solve_kwargs["max_iters"] = iters
+        problem.solve(**solve_kwargs)
+        if problem.status in ["optimal_inaccurate", "optimal"]:
+            problem_solved = True
+        elif problem.status == "unbounded_inaccurate":
+            if iters < max_iters:
+                iters *= 2
+            else:
+                raise AnalysisError(
+                    "CVXPY fit failed, probably not enough iterations for the solver"
+                )
+        elif problem.status in ["infeasible", "unbounded"]:
+            raise AnalysisError(
+                "CVXPY fit failed, problem status {} which should not happen".format(problem.status)
+            )
+        else:
+            raise AnalysisError("CVXPY fit failed, reason unknown")
+
+
+def set_default_sdp_solver(solver_kwargs: dict):
+    """Set default SDP solver from installed solvers."""
+    if "solver" in solver_kwargs:
+        return
+    if "CVXOPT" in cvxpy.installed_solvers():
+        solver_kwargs["solver"] = "CVXOPT"
+    elif "MOSEK" in cvxpy.installed_solvers():
+        solver_kwargs["solver"] = "MOSEK"
+
+
+def complex_matrix_variable(
+    dim: int, hermitian: bool = False, psd: bool = False, trace: Optional[complex] = None
+) -> Tuple[Variable, Variable, List[Constraint]]:
+    """Construct a pair of real variables and constraints for a Hermitian matrix
+
+    Args:
+        dim: The dimension of the complex square matrix.
+        hermitian: If True add constraint that the matrix is Hermitian.
+        psd: If True add a constraint that the matrix is positive
+             semidefinite.
+        trace: If True add constraint that the trace of the matrix is
```

Comment:
This reads bit strange since type hint says this is number and indeed the docs is bit unclear. What number we can set? We would set 1 for TP?

Reply:
This looks like a typo. Should be numeric value or None, not bool.